src/test/java/com/suse/saltstack/netapi/client/SaltStackClientTest.java

```diff
@@ -147,6 +155,46 @@ public void testRunRequestAsync() throws Exception {
     }
 
     @Test
+    public void testRunRequestWithSocketTimeout() throws Exception {
+        exception.expect(SaltStackException.class);
+        exception.expectMessage(containsString("Read timed out"));
+
+        // create a local SaltStackClient with a fast timeout configuration 
+        // to do not lock tests more thant 2s
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        SaltStackClient clientWithFastTimeout = new SaltStackClient(uri);
+        clientWithFastTimeout.getConfig().put(SOCKET_TIMEOUT, 1000);
+
+        stubFor(post(urlEqualTo("/login"))
+                .withHeader("Accept", equalTo("application/json"))
+                .withHeader("Content-Type", equalTo("application/json"))
+                .willReturn(aResponse()
+                .withFixedDelay(2000)));
+
+        clientWithFastTimeout.login("user", "pass");
+    }
+
+    @Test
+    public void testRunRequestWithSocketTimeoutThroughJDKConnection() throws Exception {
+        exception.expect(SaltStackException.class);
+        exception.expectMessage(containsString("Read timed out"));
+
+        // create a local SaltStackClient with a fast timeout configuration 
+        // to do not lock tests more thant 2s
+        URI uri = URI.create("http://localhost:" + Integer.toString(MOCK_HTTP_PORT));
+        SaltStackClient clientWithFastTimeout = new SaltStackClient(uri, new JDKConnectionFactory());
+        clientWithFastTimeout.getConfig().put(SOCKET_TIMEOUT, 1000);
```

Comment:
I agree and from my perspective this test is OK as it is using the 1 second timeout. It could in theory even be smaller as @isbm suggested, but IMO 1 second is small enough. The default value could maybe be verified just like this (if this is even necessary)?

    assertEquals(10000, new SaltStackClient().getConfig().get(SOCKET_TIMEOUT));